### PR TITLE
Fix Docker Mac builds

### PR DESCRIPTION
**Pull request by aquarat**
These changes will allow Airnode to build on both Mac OS arm64 *and* a Raspberry Pi.

It is possible to build both amd64 and arm64 images on one arm64 or amd64 machine using Docker BuildKit/buildx (uses qemu underneath for emulation).
It will also push the images to Docker Hub as part of this process.

I've added these buildx commands to the package.json file but it's possible that they don't belong there and instead belong in Confluence. These commands will be useful for future releases. They're here for comment, I'm happy to remove them.

I have tested the deployer on an M1 Mac.

AN-426

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -46,6 +46,11 @@
     "docker:build:artifacts": "docker rmi --force api3/airnode-artifacts && docker build --tag api3/airnode-artifacts:latest --file docker/Dockerfile .",
     "docker:build:deployer": "docker build --tag api3/airnode-deployer:latest --file packages/airnode-deployer/docker/Dockerfile .",
     "docker:build:node": "docker build --tag api3/airnode-client:latest --file packages/airnode-node/docker/Dockerfile .",
+    "docker:buildx": "docker run --privileged --rm tonistiigi/binfmt --install all && yarn docker:buildx:admin && docker:buildx:artifacts && yarn docker:buildx:deployer && yarn docker:buildx:node",
```
Comment (aquarat) [inline package.json:49]: These commands use BuildKit/buildx to build images for various architectures (amd64 and arm64 here) on the same host machine (which can be either architecture). The commands also pushe the images to Docker Hub as multi-arch compatiblle. Probably inappropriate for inclusion here, what do you think?

Comment (aquarat) [reply package.json:49]: I will remove these in favour of Confluence documentation.

File: packages/airnode-deployer/docker/Dockerfile [hunk 2]
```diff
@@ -1,7 +1,7 @@
 FROM node:14.17-alpine3.14
 
 ENV name="airnode-deployer"
-ENV terraformURL="https://releases.hashicorp.com/terraform/1.0.6/terraform_1.0.6_linux_amd64.zip" \
+ENV terraformURL="https://releases.hashicorp.com/terraform/1.0.6/terraform_1.0.6_linux_" \
```
Comment (Siegrift) [inline packages/airnode-deployer/docker/Dockerfile:4]: Why this change? Is this platform agnostic and the previous was not?

File: packages/airnode-deployer/docker/Dockerfile [hunk 3]
```diff
@@ -15,9 +15,10 @@ COPY --from=api3/airnode-artifacts /build/packages/airnode-deployer/dist ${appDi
 COPY packages/airnode-deployer/docker/entrypoint.sh /entrypoint.sh
 
     # Install external dependencies
-RUN apk add --update --no-cache su-exec git && \
+RUN export ARCH=$([ $(arch) == "x86_64" ] && echo "amd64" || echo "arm64") && \
```
Comment (Siegrift) [inline packages/airnode-deployer/docker/Dockerfile:18]: Won't this echo both `amd64` and `arm64` when arch is x86_64?

File: packages/airnode-deployer/docker/Dockerfile [hunk 4]
```diff
@@ -30,3 +31,5 @@ ENV GOOGLE_APPLICATION_CREDENTIALS=/app/gcloud/application_default_credentials.j
 WORKDIR ${appDir}
 
 ENTRYPOINT ["/entrypoint.sh"]
+
+#if [ "$(arch)" == 'aarch64' ]; then export URL="${terraformURL}arm64.zip"; else export URL="${terraformURL}amd64.zip"; fi && \
```
Comment (Siegrift) [inline packages/airnode-deployer/docker/Dockerfile:35]: Remove?

**Review by Siegrift — COMMENTED** [review]
Can someone with a macOS test this? Not sure who uses mac and has time to test...
In the worst case I can ask Michal K. to test or to borrow his laptop.

File: packages/airnode-deployer/docker/Dockerfile (hunk 2 above)
Comment (aquarat) [reply packages/airnode-deployer/docker/Dockerfile:4]: Mac OS presents in Alpine docker images on an x86_64 machine as `x86_64`, and on "M1" Macs as `aarch64` - these correspond to Terraform's `linux-amd64` and `linux-arm64` download options and also to Docker's architecture types.
Linux machines behave the same way; a Raspberry Pi running 64 bit Debian during the docker build presents as `aarch64` and `x86_64` on an x86_64 Linux host machine.

File: packages/airnode-deployer/docker/Dockerfile (hunk 3 above)
Comment (aquarat) [reply packages/airnode-deployer/docker/Dockerfile:18]: You're right, I somehow missed this, I will rectify it now. I must have only tested the previous version when I ran the arm64 deployer build.

Comment (aquarat) [reply packages/airnode-deployer/docker/Dockerfile]: Fixed, sorry about that.

**Comment by aquarat** [comment]
Ryan M indicated that he'd be interested in testing on an M1, so I'll ask him if he can help test this.

**Comment by aquarat** [comment]
The latest changes build on my Mac M1, a Raspberry Pi (64 bit) and my x86_64 machine.

Ryan M has confirmed he'll try testing it on his Mac M1 this week.

File: packages/airnode-deployer/docker/Dockerfile (hunk 2 above)
Comment (Siegrift) [reply packages/airnode-deployer/docker/Dockerfile:4]: I see - this should be named `baseTerraformURL` or something since it is an invalid URL on it's own

**Review by Siegrift — APPROVED** [review]
It would be nice to rename the terraform URL variable, but otherwise the chagnes LGTM...

Be sure to ask Ryan to test this before merging, since I cannot verify this.

**Comment by aquarat** [comment]
> It would be nice to rename the terraform URL variable, but otherwise the chagnes LGTM...

I will modify it, no problem.
> Be sure to ask Ryan to test this before merging, since I cannot verify this.

He has said he will verify it tomorrow - it'll be educational :)
